.github/workflows/post-merge-deploy-to-build.yml

```diff
@@ -0,0 +1,47 @@
+name: Package for Build
+on:
+  workflow_dispatch:
+  push:
+    branches: [main]
+
+concurrency:
+  group: post-merge-build
+  cancel-in-progress: false
+
+jobs:
+  deploy:
+    name: Package for build
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    environment:
+      name: build
+    env:
+      AWS_REGION: eu-west-2
+      ENVIRONMENT: build
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.GH_ACTIONS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: SAM Validate
+        run: sam validate --region ${{ env.AWS_REGION }} -t infrastructure/template.yaml
+
+      - name: SAM build
```

Comment:
For this step, we can use `govuk-one-login/github-actions/sam/build-application` shared action and we should then also be able to remove the `sam validate` step as the shared action does it afaik

Reply:
The same can be applied to the `dev` workflow

Reply:
I understand, this might be a c&p from the other workflow, we can update the other repository as debt rather than including this into the same debt?

Reply:
Yep makes sense - done